libs/langchain/langchain/retrievers/document_compressors/list_rerank.py

```diff
@@ -0,0 +1,99 @@
+"""Filter that uses an LLM to rerank documents listwise and select top-k."""
+from typing import Any, Callable, Dict, Optional, Sequence
+
+from langchain.callbacks.manager import Callbacks
+from langchain.chains import LLMChain
+from langchain.prompts import PromptTemplate
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+from langchain.schema import BasePromptTemplate, Document
+from langchain.schema.language_model import BaseLanguageModel
+from langchain.output_parsers import StructuredOutputParser, ResponseSchema
+
+
+def _get_default_chain_prompt() -> PromptTemplate:
+    prompt_template = """
+{context}
+Query = ```{question}```
+Sort the Documents by their relevance to the Query.
+
+{format_instructions}
+Sorted Documents:
+    """
+    response_schemas = [
+        ResponseSchema(
+            name="reranked_documents",
+            description="""Reranked documents. Format: {"document_id": <int>, "score": <number>}""",
+            type="array[dict]",
+        )
+    ]
+    output_parser = StructuredOutputParser.from_response_schemas(response_schemas)
```

Comment:
I experimented with a Pydantic parser that defines the full nested structure explicitly and saw notably more output parsing errors. Expressing the `array[dict]` type as an implicit nested type within a single `ResponseSchema` `type` argument was much more successful.